src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.cs

```diff
@@ -318,6 +324,24 @@ public Task<PingReply> SendPingAsync(IPAddress address, int timeout, byte[] buff
             return SendPingAsyncInternal(address, timeout, buffer, options);
         }
 
+        public Task<PingReply> SendPingAsync(IPAddress address, TimeSpan timeout, byte[]? buffer = null,
+            PingOptions? options = null, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
This needs to be:
```C#
if (cancellationToken.ThrowIfCancellationRequested)
{
    return Task.FromCanceled<PingReply>(cancellationToken);
}
```
Same for the other case below.  Cancellation exceptions should not be thrown synchronously out of async methods.

Reply:
https://github.com/dotnet/runtime/pull/64860#discussion_r836496120